packages/react/src/components/DataTable/stories/DataTable-sorting.stories.js

```diff
@@ -66,6 +66,40 @@ export const Default = () => (
   </DataTable>
 );
 
+export const WithInitialState = (args) => (
+  <DataTable
+    initialSortColumn="name"
```

Comment:
This seems like a nice opportunity to encourage the use of the headers data itself rather than a hardcoded header key string.

```suggestion
    initialSortColumn="headers[0].key"
```